src/service/repo/signedMessageRepo.ts

```diff
@@ -162,14 +166,29 @@ async function createSignedChat(members: string[]) {
   }
 }
 
+async function getSignedChannelDetail(channelType: GetstreamChannelType, channelId: string) {
+  try {
+    const response = await api.get(baseUrl.getSignedChannelDetail(channelType, channelId));
+    if (response.status === 200) {
+      return Promise.resolve(response.data?.data);
+    }
+
+    return Promise.reject(response.data?.status);
+  } catch (e) {
+    console.log(e);
+    return Promise.reject(e);
+  }
+}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `getSignedChannelDetail` function, error handling could be improved. Currently, if an error occurs during the API call, it's logged to the console but not thrown again. This could make debugging difficult because the calling code might not realize an error occurred. Consider rethrowing the error after logging it.
Also, it's generally not a good idea to reject a promise with a non-error object (`response.data?.status`). Instead, consider creating a new Error instance with a meaningful message.
```diff
- return Promise.reject(response.data?.status);
+ return Promise.reject(new Error(`Failed to fetch signed channel detail. Status: ${response.data?.status}`));
```
And in the catch block:
```diff
- console.log(e);
- return Promise.reject(e);
+ console.error(e);
+ throw e;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->